### PR TITLE
Fix/obj setting

### PR DESCRIPTION
**Pull request by PingHsunTsai**
### What type of change is this?
#183

- [ ] Added `DefaultLayout` to handle gerneral `layout` setting to minimal.
- [ ] Added `TextEdit` to handle `name` change.
- [ ] Added `Observer` to handle `is_selected` state update or more.

### CODE REVIEW

**Comment by PingHsunTsai** [comment]
The modified obj attributes are able to save to scene, but there is a problem to load the color attribute from scene, we can fo it in another PR

**Comment by Licini** [comment]
This PR is now mixed with fixing the layout of object settings, and implementation of `Observer` which we might need more iterations vs `Signal` pattern. Can you isolate out the `Observer` part?

**Comment by PingHsunTsai** [comment]
Yes, but the ability to open ObjectSetting from Sceneform selection will not be available for now(just a heads up).
I will try is there another quick solution.

**Comment by Licini** [comment]
How about add the object settings cmd as callback function in the config of `SceneForm`?
```python
{
    "type": "Sceneform",
    "columns": [
        {"title": "Name", "type": "label", "text": lambda obj: obj.name},
        {"title": "Show", "type": "checkbox", "checked": lambda obj: obj.show, "action": lambda obj, checked: setattr(obj, "show", checked)},
    ],
    "on_select": obj_settings_cmd # something like this
},

```

**Comment by PingHsunTsai** [comment]
yes on it!

**Comment by Licini** [comment]
> How about add the object settings cmd as callback function in the config of `SceneForm`?
> 
> ```python
> {
>     "type": "Sceneform",
>     "columns": [
>         {"title": "Name", "type": "label", "text": lambda obj: obj.name},
>         {"title": "Show", "type": "checkbox", "checked": lambda obj: obj.show, "action": lambda obj, checked: setattr(obj, "show", checked)},
>     ],
>     "on_select": obj_settings_cmd # something like this
> },
> ```

Just realized there was already a `callback` parameter in `SceneForm` which does exactly this, you can use that directly. Let's just use name `callback` instead of `on_select`

**Comment by Licini** [comment]
Some observations here:
1. the object setting form should hide when nothing selected
image

2. Name should not be number 
image

3. Color picker is too thin in vertical direction 
image

4. The object form should be in a scrollable, right now the form has a fixed size, the portion between scene form and object form should be adjustable

**Comment by PingHsunTsai** [comment]
regarding your first point @Licini  
I think Tom prefer to have a placeholder there, but I just added  `self.hide_widget = True` option in `SideBarRight`, so this can be decided by user as well.
For the scrolling space, it needs to be restructure a bit, cus it takes different type than layout, so I will propose we do it in another PR
The rest I will try to adjust them now.

**Comment by PingHsunTsai** [comment]
Hi @Licini and @tomvanmele this PR is ready!

File: src/compas_viewer/components/objectsetting.py [hunk 1]
```diff
@@ -86,11 +51,23 @@ class ObjectSetting(QWidget):
...
     update_requested = Signal()
 
-    def __init__(self, viewer: "Viewer"):
+    def __init__(self, viewer: "Viewer", items: list[dict]):
         super().__init__()
         self.viewer = viewer
-        self.layout = QVBoxLayout(self)
-        self.spin_boxes = {}
+        self.items = items
+        self.setFixedHeight(200)
```
Comment (Licini) [inline src/compas_viewer/components/objectsetting.py:58]: We should not set a fix height, the form should be allowed to adjusted vertically like this:
(also can you make the items vertically align to the top?)
image

**Comment by Licini** [comment]
Noticed some issues with spinbox:
1. Plz make the spinbox wider and show the digits after
2. use 1 decimal place instead of two
3. Make the step 0.1 instead of 1, right now if I click down arrow it set to value 0 and start to throw render errors. (linewidth value can not <= 0) .
image

**Review by Licini — CHANGES_REQUESTED** [review]
Looking much better! but still plz make the changes mentioned above

**Comment by Licini** [comment]
one more question, how do you disable the object settings?

**Comment by PingHsunTsai** [comment]
@Licini https://github.com/compas-dev/compas_viewer/pull/181/commits/d9369e2bbe5da214fba75623c3bab9dd1dfbc483
I just add a setter here

**Comment by Licini** [comment]
Thanks @PingHsunTsai ! Merging Finally!

**Pull request merged**